Add climate baseline verification script

rod-glover opened the pull request:
Verification script: `scripts/verify-load-climate-baseline-values.py`

This script is to be run after running the load script `scripts/load-climate-baseline-values.py`. It checks the database to make sure that the load script actually did what we wanted it to do. The verification script raises an exception where it finds any discrepancy.

rod-glover commented:
This script may not quite be finished: I am awaiting some final information from Faron on certain data (precip values not from rainfall) that should be either loaded or filtered out of queries against the climatology datasets. This may affect the verification script.

jameshiebert commented on pycds/climate_baseline_helpers.py:
```diff
@@ -177,3 +177,130 @@ def parse_line(line):
+        ('Tn_Climatology', 'minimum', 'Min.'),
+    ]:
+        temp_variable = expect_variable(name)
+        expect_variable_attr(temp_variable, 'unit', u'celsius')
```
You shouldn't need to use ``u''`` style unicode strings. Those were a vestige of the first couple minor versions of Python 3 (3.0, 3.1) that everyone realized was a mistake and never used again. E.g.

```python
$ python2.7
Python 2.7.12 (default, Nov 29 2016, 16:30:36) 
[GCC 4.9.3] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> u'foo' == 'foo'
True
$ python3.3
Python 3.3.5 (default, May 10 2016, 10:39:16) 
[GCC 4.9.3] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> u'foo' == 'foo'
True
```

jameshiebert replied:
You really could just use a dict here to eliminate a lot of redundancy:

```python
expected_attrs = {'unit': 'celcius', 'standard_name': 'air_temperature'}
for key, val in expected_attrs.items():
    expect_variable_attr(temp_variable, key, val)
```

rod-glover replied:
Duh. Thanks.

jameshiebert commented on scripts/verify-load-climate-baseline-values.py:
```diff
@@ -0,0 +1,111 @@
+
+    verify_baseline_network_and_variables(session)
+
+    def convert_temp_data(temps_in_10thsC):
```
While it's somewhat obvious from reading the code, you may want to state your assumptions about the input data. "Temperatures are coded as integers in 10ths of a degree and the special integer -9999 represents a missing value". History shows that it's virtually guaranteed that these assumptions will change at some point and someone will try and run this same code with different units :D

jameshiebert commented on scripts/verify-load-climate-baseline-values.py:
```diff
@@ -0,0 +1,111 @@
+        },
+    }
+
+    # Quis custode custodes?
```
Is this French?

jameshiebert commented on tests/conftest.py:
```diff
@@ -17,6 +17,27 @@
     logging.basicConfig(level=logging.DEBUG, stream=sys.stdout)
 
+
+@fixture(scope='session')
```
Ah, this is slick using the engine fixture at session scope and the session fixture at module scope. Good, good.

jameshiebert commented on tests/test_climate_baseline_helpers.py:
```diff
@@ -45,10 +43,19 @@ def histories(stations):
-    def test_creates_the_expected_new_network_record(empty_database_session):
-        sesh = empty_database_session
+    def test_creates_the_expected_new_network_record(session):
+        sesh = session
```
Eh, you barely use the local variable sesh... 1-2x per test. Is it really worth the aliasing? Likewise for the next several tests.

rod-glover commented:
Ready for next review

rod-glover commented:
File and variable options were an incorrect cut-paste from the loading script. We _could_ just verify one variable at a time, but that's not what I've coded and I think unnecessary.

I was wondering about the single-hyphen, two-char option: I occasionally but rarely see it done. I can remove the "c"s from each if you like.

rod-glover commented:
Ready for next review.

I linted many of my files and fixed PEP8 compliance problems. You should see less red text. Unfortunately, I changed some files that were not originally involved in this particular branch. But you should be able to spot these easily.
